packages/schematics/src/command-line/affected-apps.ts

```diff
@@ -1,29 +1,25 @@
-import * as ts from 'typescript';
-import * as path from 'path';
-import { normalizedProjectRoot, ImplicitDependencies } from './shared';
+import { ImplicitDependencies, mtime, readDependencies } from './shared';
+import { DepsCalculator, Dependency, Deps } from './deps-calculator';
+import * as appRoot from 'app-root-path';
+import { existsSync } from 'fs';
+import { fileExists } from '../utils/fileutils';
 
 export enum ProjectType {
   app = 'app',
   e2e = 'e2e',
   lib = 'lib'
 }
 
-export enum DependencyType {
-  es6Import = 'es6Import',
-  loadChildren = 'loadChildren',
-  implicit = 'implicit'
-}
-
 export type ProjectNode = {
   name: string;
   root: string;
+  mtime: number;
```

Comment:
I think we should use a single mtime for the workspace unless it doesn't allow to implement certain scenarios. We can use the mtime of the nxdeps file itself for that.

Why:
nxdeps always stores a consistent view of the repo. If we write nxdeps at time T1, then it stores a correct view of what depends on what at T1. If we then change any file in the repo say at T2,  we just need to check if T2 > T1, and if it is, recompute incrementally the nodes touched by us. The fact that the changed project was updated long time ago (T0 and not T1) is not relevant.

Reply:
These are `mtime`s of the projects at the time of calculation and these `mtime`s are used to determine if they need to be processed (https://github.com/nrwl/nx/pull/923/files#diff-1ff6608f166d6a9bb3dcdb5b07214c48R229). I think it is possible to decrease our granularity to the file level but that requires some further work to store the imports of the file.

My initial try was to use each file mtime but it was flawed because while it could add a non-existing edge, it was not enough information to determine if we could remove existing edges.

Reply:
Hiding because implementation has changed.